### PR TITLE
[kostal] Kostal Inverter Piko New Generation Binding initial contribution

### PR DESCRIPTION
**Pull request by basse04**
[kostalinverterpikonewgeneration]
Initial contribution

The binding  scrape the inverter of type Kostal Piko New Generation and pulls the scraped values to respectiv channel.
The binding is build for bnd-system with Eclipse 19-3.
The most of channels are for reading, but some are for writing as described in README.md

### CODE REVIEW

**Comment by basse04** [comment]
Also-by: Christian Schneider <>
Also-by: Örjan Backsell <orjan.backsell@gmail.com>
Signed-off-by: Örjan Backsell <orjan.backsell@gmail.com>

**Comment by basse04** [comment]
Hi,

I can't find any [Error] / [Warning] regarded to Kostal in the log, maybe I can't somehow  see it.

The only text I can find is like:

[INFO] openHAB Add-ons :: Bundles :: Kostal Binding ....... SUCCESS [ 17.073 s]

Please, help me if there is something I have misunderstood regarding to the logfile.

Br Basse

**Comment by Hilbrand** [comment]
It doesn't build your binding because it's not in the bundle pom. It only reports about building the existing binding.

I see you have greatly improved  up on #4748. If I have it correctly this is for the second generation inverters. Currently the 1st and 3rd generation are both part of the kostalinverter binding, each in a separate package in the binding. I would like to suggest to move this binding in the kostalinverter binding under it's specific package and use more clearer product names for thing names. By having it's own package means it doesn't require a lot of integration work as it's mostly separated.

**Comment by basse04** [comment]
Thank's, 
yes this one is for Kostal Piko 2'nd generation (described as New generation).
Do you like me to move this one  into the Kostal Inverter binding ?, if so I will have a look asap. Just let me know what you think will be the best way to integrate.
Just another question, which version of Eclipse is best to use right now, I'm using 19-3?

**Comment by Hilbrand** [comment]
Yes please do move with the kostal inverter binding. If you put it in a specific package like the other 2 it should not have much interference, like with the other 2.
See if you can use the actual product code for the things like with the other, this is clearer than using 'new generation's as that is very generic and might not age well if there will be new products. You can put the thing definitions in a separate things xml, but see if you can reuse channel definitions of the other things.
All recent eclipse versions work, 19-3 is fine, I also use it.

**Comment by basse04** [comment]
I will do my very best, and Thanks for the advices.
I let you know about the progress asap.

**Comment by basse04** [comment]
Hi,
I am working to integrate the two bindings. I can so far have the new generation part running inside the Kostal Inverter binding , I can see it in PaperUI /Things, I can scrape the values and also see them in Paper UI / Control.
But I have not managed to view this second generation in Paper UI / Configurations / Bindings / Kostal Inverter Binding / Supported Things.
If I swap bindingId  in ESH-INF/thing/PIKO1020.xml can i either see it in SupportedThings but then will not the binding work or reversed.
Do you maybe have some suggestions ?

**Comment by Hilbrand** [comment]
@basse04 you probably need to align the bindingId in your things files, and in your code to match the bindingId of the original binding. If you show me your code I can give more detailed feedback.,

**Comment by basse04** [comment]
Thank's
I put the code on https://github.com/basse04/Kostal-Test
I will change names of channels etc. but this is the first release so far

**Comment by basse04** [comment]
Forgot, with this bindingId in PIKO1020.xml will the binding start and scrape and show values in PaperUI but the binding will not show up in Supported Things

**Comment by basse04** [comment]
Hi,
I have uploaded the latest integrated version on https://github.com/basse04/Kostal_Second_Generation. I had done some mismatches between names, so Thank's for the help.
Please have a look if you think that this could be OK regarded to Channel names etc.

//Basse

File: org.openhab.binding.kostalinverter/.settings/org.eclipse.core.resources.prefs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+eclipse.preferences.version=1
+encoding//src/main/java=UTF-8
```
Comment (cpmeister) [inline org.openhab.binding.kostalinverter/.settings/org.eclipse.core.resources.prefs:2]: remove this file, it shouldn't be included in this PR.

File: org.openhab.binding.kostalinverter/.settings/org.eclipse.jdt.core.prefs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+eclipse.preferences.version=1
```
Comment (cpmeister) [inline org.openhab.binding.kostalinverter/.settings/org.eclipse.jdt.core.prefs:1]: remove this file

File: org.openhab.binding.kostalinverter/.settings/org.eclipse.m2e.core.prefs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
```
Comment (cpmeister) [inline org.openhab.binding.kostalinverter/.settings/org.eclipse.m2e.core.prefs:1]: remove this file

**Comment by cpmeister** [comment]
Your changes don't show up in this PR. And I can't merge this binding without the changes...
Maybe you should try resubmitting this PR using your current working branch so we can get this resolved.

**Comment by cpmeister** [comment]
I think I know what is going on. You need to make a fork of the `openhab-addons` repo and then add your code in there then submit a new PR. I should be able to review it just fine when you do that.

**Comment by cpmeister** [comment]
This PR was replaced by #7334
